Clean-up wording in various error messages

dsas opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of DOTCOM-13092

## Proposed Changes

 * Remove cutsey wording like 'uh oh' from error messages using the <EmptyContent> component.
 * Remove cartoony illustrations on empty screens
 * Remove the default illustration from EmptyContent
 * Use the brand font for the title

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

We want to change the image we're presenting with some of our in-product messaging. We've focused here on the EmptyContent component. EmptyCOntent is used to display full screen "empty" messages - think errors, prompts to add content, 404s etc. They are often paired with old cartoony illustrations, which isn't the look we're going for right now.

The EmptyContent component does still retain support for displaying an image though - sometimes it is genuinely useful, like when displaying theme screenshots.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

There are dozens :) This is just a random sample:

### [SSO with invalid URL](https://github.com/Automattic/wp-calypso/pull/103372/files#diff-4f58236c96528c78f9053d39ca2d0d4853e11daa4ad036818e3e27b231d65019)
 1. Log out (probably)
 2. Visit a malformed Jetpack SSO URL like /jetpack/sso/:siteid (i.e. omit the nonce at the end)

Before | After
-------|------
 <img width="1127" alt="Screenshot 2025-05-14 at 09 29 34" src="https://github.com/user-attachments/assets/db579ef4-2da5-448d-8787-ccfba9e9ddc3" />|<img width="1127" alt="Screenshot 2025-05-14 at 09 29 58" src="https://github.com/user-attachments/assets/9c69751c-0fd2-41fe-855a-c7f9a21266b9" />



 ### [Subscriptions list](https://github.com/Automattic/wp-calypso/pull/103372/files#diff-94d0a0cff97e9123fa5f3c0a2da76895edf60b04a44e7b9ec8723844698b7531)

 1. Choose a free site
 4. Go to /purchases/subscriptions/:siteSlug

Before | After
-------|------
<img width="1127" alt="Screenshot 2025-05-13 at 15 28 33" src="https://github.com/user-attachments/assets/a07e5da2-4326-48e3-a72a-b4e643458eb4" /> | <img width="1127" alt="Screenshot 2025-05-14 at 09 14 26" src="https://github.com/user-attachments/assets/c17d23dc-f9ed-4ab3-9d11-98c679ae0818" />

### [Community theme installation](https://github.com/Automattic/wp-calypso/pull/103372/files#diff-cd2580eb28ed5651c9598170dff2835bad442f70c8a3305fce37268b75c219d4)
 1. Find a simple site with the business plan
 2. Go to /marketplace/theme/vividmono/install/:siteSlug where vividmono is the slug of a community theme
 3. Note the illustration is retained - it's a useful reminder of what the theme looks like

Before | After
-------|------
 <img width="1018" alt="Screenshot 2025-05-14 at 09 42 16" src="https://github.com/user-attachments/assets/d283c99b-7f7a-4208-a4bb-3f0c0cad3012" /> | <img width="1018" alt="Screenshot 2025-05-14 at 09 41 40" src="https://github.com/user-attachments/assets/6cd85c75-6054-4530-a0a7-aed8174d65f1" />


### [Mailboxes](https://github.com/Automattic/wp-calypso/pull/103372/files#diff-6c1dab58376a89002351477f619fad03806d73e077120bf12b01b451b2cca3fb)

 1. Go to /mailboxes/:someP2Domain

Before | After
-------|------
<img width="992" alt="Screenshot 2025-05-14 at 17 21 53" src="https://github.com/user-attachments/assets/370a1626-5937-47e9-b325-b8f74016a357" /> | <img width="992" alt="Screenshot 2025-05-14 at 17 22 24" src="https://github.com/user-attachments/assets/56220d3c-fca6-4910-9628-4fe25f1d1139" />

### [Pattern library category search](https://github.com/Automattic/wp-calypso/pull/103372/files#diff-73a5e0a421d2d5eaa06fea1244fd04e135da6b4f813a7c2abcff5965f4237ec2)

1. Go to /patterns/cider

Before | After
-------|------
<img width="1108" alt="Screenshot 2025-05-14 at 17 35 44" src="https://github.com/user-attachments/assets/96bf0c3a-9a23-401f-9e21-0ffdc5c45767" /> | <img width="1108" alt="Screenshot 2025-05-14 at 17 35 52" src="https://github.com/user-attachments/assets/4fffff52-88cb-4a25-bacd-cdcdbebafc15" />

### [Invite other users](https://github.com/Automattic/wp-calypso/pull/103372/files#diff-4fe182b4f7dd048fa3c64d78957f2ba7231ca95f30a94d0d4aa0a37f508b353b)

 1. Be a non-admin user on a site
 2. Go to /people/new/siteSlug.wordpress.com

Before | After
-------|------
<img width="1108" alt="Screenshot 2025-05-14 at 17 39 24" src="https://github.com/user-attachments/assets/46241fb7-5cbb-4924-b4a8-1b8f8988d4e1" /> | <img width="1108" alt="Screenshot 2025-05-14 at 17 40 03" src="https://github.com/user-attachments/assets/c9e6b383-beff-4a4b-9974-ecd8eee55755" />

### [Reader - not posted yet](https://github.com/Automattic/wp-calypso/pull/103372/files#diff-d4d2beec71279a680a4194c5034ea2d75be83bcdcf0381dca12a1e4b9ea945b3)

 1. Find a site with no posts
 2. Go to /reader/blogs/:siteSlug.wordpress.com
 3. Notice it looks mildly broken in prod too.

Before | After
-------|------
<img width="1108" alt="Screenshot 2025-05-14 at 18 01 46" src="https://github.com/user-attachments/assets/d157cdef-a467-483c-9f0f-fcfdc5918dd5" /> | <img width="1108" alt="Screenshot 2025-05-14 at 18 02 05" src="https://github.com/user-attachments/assets/58167bcb-1534-4739-ae3e-e91a401f7879" />

### [Marketing traffic - unauthorised](https://github.com/Automattic/wp-calypso/pull/103372/files#diff-5dee8efb993f0a228278438c1745593de4546dac3ef4422d2efd3c39966e6c2b)

1. Be a contributor (not an admin) on a site
2. Go to /marketing/traffic/:siteSlug.wordpress.com

Before | After
-------|------
<img width="1108" alt="Screenshot 2025-05-14 at 18 05 02" src="https://github.com/user-attachments/assets/e65652ba-e942-4107-90c3-57c7991f37ae" />| <img width="1108" alt="Screenshot 2025-05-14 at 18 05 09" src="https://github.com/user-attachments/assets/fc874b5e-302e-42ed-b8e5-0889e1cbe30e" />

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

dsas committed:
Clean-up wording in various error messages

This removes cutsey wording like 'uh oh' from error messages using the
<EmptyContent> component.

dsas commented:
Opened a draft PR with the text changes i'm proposing. Will work further on illustrations etc.

fditrapani reviewed (approved):
Thanks for these updates @dsas. These all read well. Looks good on my end!

dsas committed:
Remove cartoony illustrations on empty screens

The EmptyContent component is used to display full screen "empty"
messages - think errors, prompts to add content, 404s etc. They are
often paired with old cartoony illustrations, which isn't the look we're
going for right now.

The EmptyContent component does still retain support for displaying an
image though - sometimes it is genuinely useful, like when displaying
theme screenshots.

dsas committed:
Fix unit test

dsas committed:
fix linter

dsas committed:
another linter fix

dsas committed:
til we lint markdown files

dsas committed:
Use the brand font

dsas commented on client/my-sites/media-library/list-no-content.jsx:
untangled so is no longer reachable

dsas commented on client/my-sites/marketing/do-it-for-me/difm-lite-in-progress.tsx:
We don't offer the DIFM service any longer...

dsas replied:
Though this might be difm express which we do still offer...

dsas commented on client/my-sites/media-library/list-plan-promo.js:
untangled so is no longer reachable

dsas commented on client/my-sites/pages/page-list.jsx:
untangled so is no longer reachable

dsas commented on client/my-sites/post-type-list/empty-content.jsx:
untangled, don't think it's available

dsas commented:
Removed the auto-added CCs, as I _think_ it's overkill for so many people to be involved. Hopefully this doesn't come back to bite me.

xavier-lc reviewed (approved):
Changes LGTM 👌

Have you considered getting rid of the now unused images? I haven't checked all, but for example `static/images/illustrations/page-bag.svg` and `static/images/illustrations/illustration-seller.svg` aren't used anywhere after your changes.

dsas commented:
> Have you considered getting rid of the now unused images? I haven't checked all, but for example static/images/illustrations/page-bag.svg and static/images/illustrations/illustration-seller.svg aren't used anywhere after your changes.

I thought I had checked them and they were in use, I’ll check them again because obviously I missed at least one.

dsas committed:
Removed some now unused svgs

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17489713

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @dsas for including a screenshot in the description! This is really helpful for our translators.
